apps/extension/src/routes/popup/settings/settings-connected-sites.tsx

```diff
@@ -0,0 +1,122 @@
+import { Link1Icon, LinkBreak1Icon, MagnifyingGlassIcon, TrashIcon } from '@radix-ui/react-icons';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Button, FadeTransition, Input } from '@penumbra-zone/ui';
+import { LinkGradientIcon } from '../../../icons';
+import { SettingsHeader } from '../../../shared';
+import { OriginRecord, localExtStorage } from '@penumbra-zone/storage';
+import { DisplayOriginURL } from '../../../shared/components/display-origin-url';
+import Map from '@penumbra-zone/polyfills/Map.groupBy';
+
+export const SettingsConnectedSites = () => {
+  const [search, setSearch] = useState<string>();
+
+  const [sitesFromStorage, setSitesFromStorage] = useState<OriginRecord[]>();
+
+  const loadSitesFromStorage = useCallback(
+    async () => setSitesFromStorage(await localExtStorage.get('connectedSites')),
+    [],
+  );
+  useEffect(() => void loadSitesFromStorage(), [loadSitesFromStorage]);
+
+  const discardSiteRecord = useCallback(
+    (site: OriginRecord) => {
+      if (!sitesFromStorage) return;
+      const sitesWithoutRecord = sitesFromStorage.filter(({ origin }) => origin !== site.origin);
+      void localExtStorage.set('connectedSites', sitesWithoutRecord);
+      setSitesFromStorage(sitesWithoutRecord);
+    },
+    [sitesFromStorage],
+  );
+
+  const filteredByAttitude = useMemo(
+    () =>
+      Map.groupBy(
+        (sitesFromStorage ?? []).filter(site => !search || site.origin.includes(search)),
+        ({ attitude }) => attitude,
+      ),
+    [sitesFromStorage, search],
+  );
```

Comment:
All this logic feels like it would be better in a zustand store. Can we migrate this?

Reply:
we need the filtering in-page as the user types in the search box. is it reasonable to round-trip to store for that?

Reply:
It's less a roundtrip thing and more a "business logic goes in the state manager" kind of thing. Something more complicated than a few hooks should probably be migrated out.